src/Aspire.Hosting/Publishing/DockerContainerRuntime.cs

```diff
@@ -58,4 +59,45 @@ public async Task BuildImageAsync(string contextPath, string dockerfilePath, str
             throw new DistributedApplicationException($"Docker build failed with exit code {exitCode}.");
         }
     }
+
+    public Task<bool> CheckIfRunningAsync(CancellationToken cancellationToken)
```

Comment:
Not blocking feedback but do we want to add more granularity here? For example, if the container runtime is running and not healthy we want to surface why in the logs.

I realize this might be expose the same "you have to grep the logs to find the error" problem but something to think about.